bmc-acf/acf_manager.cpp

```diff
@@ -0,0 +1,461 @@
+        {
+            // Get the size of the file
+            sInputFile.seekg(0, std::ios::end);
+            std::streampos size = sInputFile.tellg();
+            sInputFile.seekg(0, std::ios::beg);
+
+            bufferParm.reserve(size);
+            bufferParm.assign(size, 0);
+
+            sInputFile.read((char*)bufferParm.data(), size);
+            sInputFile.close();
+
+            return true;
+        }
+        else
+        {
+            log<level::ERR>("Failed to open file ");
+        }
+    }
+    else
+    {
+        log<level::ERR>("filename empty");
+    }
+
+    return false;
+}
+
+std::string readMachineSerialNumberProperty(const std::string& obj,
+                                            const std::string& inf,
+                                            const std::string& prop)
+{
+    std::string propSerialNum = "";
+    std::string object = obj;
+    auto bus = sdbusplus::bus::new_default();
+    auto properties = bus.new_method_call(
+        "xyz.openbmc_project.Inventory.Manager", object.c_str(),
+        "org.freedesktop.DBus.Properties", "Get");
+    properties.append(inf);
+    properties.append(prop);
+    try
+    {
+        auto result = bus.call(properties);
+        if (!result.is_method_error())
+        {
+            std::variant<std::string> val;
+            result.read(val);
+            if (auto pVal = std::get_if<std::string>(&val))
+            {
+                propSerialNum.assign((pVal->data()), pVal->size());
+            }
+            else
+            {
+                log<level::ERR>("could not get the host's serial number\n");
+            }
+        }
+    }
+    catch (const std::exception& exc)
+    {
+        log<level::ERR>("dbus call for getting serial number failured: ");
+        propSerialNum = "";
+    }
+    return propSerialNum;
+}
+
+int readFieldModeProperty(const std::string& obj, const std::string& inf,
+                          const std::string& prop)
+{
+    bool propBool = false;
+    std::string object = obj;
+    auto bus = sdbusplus::bus::new_default();
+    auto properties = bus.new_method_call(
+        "xyz.openbmc_project.Software.BMC.Updater", object.c_str(),
+        "org.freedesktop.DBus.Properties", "Get");
+    properties.append(inf);
+    properties.append(prop);
+    try
+    {
+        auto result = bus.call(properties);
+        if (!result.is_method_error())
+        {
+            std::variant<bool> val{false};
+            result.read(val);
+            if (auto pVal = std::get_if<bool>(&val))
+            {
+                log<level::ERR>(
+                    "dbus call for getting FieldModeProperty failured: ");
+                propBool = (*pVal);
+            }
+        }
+    }
+    catch (const std::exception& exc)
+    {
+        log<level::ERR>("dbus call failure ");
+        return FAILURE;
+    }
+
+    return (int)propBool;
+}
+
+CeLogin::CeLoginRc
+    verifyAcfSerialNumberAndExpiration(std::vector<uint8_t>& accessControlFile,
```

Comment:
can this be static class method so that it is in the scope of the class and also add documentation what this method does?

same for all the methods.

Reply:
Yes. Changed that.